### DISCUSSION TITLE
Changing RWS listening port on real controller

### DISCUSSION THREAD

**Discussion by Amin173** [ros-industrial/abb_robot_driver#45] [Q&A]
This isn't exactly an issue with the repository but I was wondering if someone could help me with changing the RWS listening port on a real controller. The default port in the [rws_service_provider](https://github.com/ros-industrial/abb_robot_driver/blob/master/abb_rws_service_provider/launch/rws_service_provider.launch#:~:text=%3Carg%20name%3D%22robot_port%22%20%20%20%20%20%20%20%20%20%20%20%20default%3D%2280%22%20%20%20%20doc%3D%22Port%20number%20of%20the%20robot%20controller%27s%20RWS%20server%22/%3E) and [rws_state_publisher](https://github.com/ros-industrial/abb_robot_driver/blob/master/abb_rws_state_publisher/launch/rws_state_publisher.launch#:~:text=%3Carg%20name%3D%22robot_port%22%20%20%20%20%20%20%20%20%20%20%20%20default%3D%2280%22%20%20%20%20doc%3D%22Port%20number%20of%20the%20robot%20controller%27s%20RWS%20server%22/%3E) launch files is set to 80, but in my tests, my robot's default port was 443. I found [this post](https://forums.robotstudio.com/discussion/12177/how-to-change-the-listening-port-of-the-virtual-controller-robotware-6-x-and-7-x) for changing the default port on a virtual controller (robotstudio), but wasn't able to find the instructions for doing the same on a real controller.
Thank you!

**Comment by gavanderhoorn** [comment]
I can't answer your question, but just wanted to ask whether you're using an OmniCore controller or a regular IRC5.

`443` implies SSL/TLS encrypted HTTP, which I've not seen mentioned in RW6/IRC5 documentation IIRC.

**Comment by Amin173** [comment]
Thanks for your response. I am using an OmniCore controller with a modified version of this driver which is RWS2 compatible.
